assets/js/data/cart/push-changes.ts

```diff
@@ -126,9 +127,11 @@ const updateCustomerData = debounce( (): void => {
  */
 export const pushChanges = (): void => {
 	const store = select( STORE_KEY );
+	const hasValidationErrors =
+		select( VALIDATION_STORE_KEY ).hasValidationErrors();
 	const isInitialized = store.hasFinishedResolution( 'getCartData' );
 
-	if ( ! isInitialized ) {
+	if ( ! isInitialized || hasValidationErrors ) {
```

Comment:
We're using validation errors at multiple places. If there is a validation error in the name field and if a user decides to change the email or address instead of fixing the name field error, I think this condition will prevent communicating the address or other changes to the server. Do you think we should prevent communicating to the server for other fields as well until all validation errors are resolved? This could be the edge case. Please correct me if I'm mistaken.

Reply:
Steps to reproduce: 

1. Go to the Checkout block. 
2. Enter an invalid email address. 
3. Notice you get the validation error. 
4. Enter an invalid zip code without fixing the email address error. 
5. Try filling in the other fields. Notice you won't get notified for the invalid zip code. 
6. Fix the email address error. 
7. Notice zip code errors pop up. 

I think we should display all the errors immediately once we leave any field. Please let me know your thoughts.

Reply:
@tarunvijwani blocking updates is the intention here. The fix for your case would be to add client-side validation for the ZIP code field too. We should try to avoid reliance on the server for field validation if possible.

Reply:
Another thing here is that we should somehow allow partial updates, if email is filled and is valid it should push, not sure if it's in this PR.